async_processor/sidecar/sidecar.py

```diff
@@ -86,6 +87,21 @@ async def iter(self) -> AsyncIterator[StreamBatch]:
             raise self._fetcher_task.exception()
 
 
+def _resolve_url(input_message: InputMessageInterface) -> str:
+    url_from_message = input_message.get_request_url()
+
+    if not url_from_message:
+        return settings.destination_url
+
+    parsed_base_url = urlparse(settings.destination_url)
+    parsed_url_from_message = urlparse(url_from_message)
+
+    return parsed_url_from_message._replace(
+        netloc=parsed_base_url.netloc,
+        scheme=parsed_base_url.scheme,
+    ).geturl()
```

Comment:
Is this like a security feature to not allow sending  request to arbitrary hosts?

Reply:
Let me discuss this offline with you.